packages/browser-wallet/src/popup/pages/AddTokens/AddTokens.tsx

```diff
@@ -0,0 +1,190 @@
+    return (
+        <div className="tokens__add__element" title={metadata.description}>
+            {metadata.display?.url && (
+                <img alt={metadata.name} className="tokens__add__token-display" src={metadata.display.url} />
+            )}
+            <p>{metadata.name}</p>
+        </div>
+    );
+}
+
+/**
+ * Component used to pick token from a CIS-2 compliant smart contract instance.
+ */
+function PickTokens({
+    contractDetails,
+    onFinish,
+    defaultTokens,
+}: {
+    contractDetails: ContractDetails;
+    onFinish: (updatedTokenList: TokenIdAndMetadata[]) => void;
+    defaultTokens: TokenIdAndMetadata[];
+}) {
+    const { t } = useTranslation('account', { keyPrefix: 'tokens' });
+    const client = useAtomValue(jsonRpcClientAtom);
+    const [accountTokens, setAccountTokens] = useState<TokenIdAndMetadata[]>(defaultTokens);
+    const addToast = useSetAtom(addToastAtom);
+    const form = useForm<FormValues>();
+
+    const onSubmit: SubmitHandler<FormValues> = async ({ id }) => {
+        if (accountTokens.some((token) => token.id === id)) {
+            addToast(t('duplicateId'));
+            return;
+        }
+        const tokenUrl = await getTokenUrl(client, id || '', contractDetails);
+        const meta = await getTokenMetadata(tokenUrl);
+        setAccountTokens((tokens) => [...tokens, { id, metadata: meta, metadataLink: tokenUrl }]);
+    };
+
+    return (
+        <div className="tokens__add__container">
+            <UncontrolledInput
+                readOnly
+                className="tokens__add__input"
+                label={t('contractIndex')}
+                value={contractDetails.index.toString()}
+            />
+            <UncontrolledInput
+                readOnly
+                className="tokens__add__input"
+                label={t('contractName')}
+                value={contractDetails.contractName}
+            />
+            <Form formMethods={form} className="tokens__add__add-token" onSubmit={onSubmit}>
+                {(f) => (
+                    <>
+                        <Input register={f.register} label={t('tokenId')} name="id" />
+                        <Submit>{t('addToken')}</Submit>
+                    </>
+                )}
+            </Form>
+            {accountTokens.map((token) => (
+                <DisplayToken key={token.id} metadata={token.metadata} />
+            ))}
+            <Button onClick={() => onFinish(accountTokens)} className="tokens__add__submit">
+                {t('updateTokens')}
+            </Button>
+        </div>
+    );
+}
+
+export default function AddTokens() {
+    const [contractDetails, setContractDetails] = useState<ContractDetails>();
+    const [tokenMetadata, setTokenMetadata] = useAtom(tokenMetadataAtom);
+    const [tokens, setTokens] = useAtom(storedTokensAtom);
+    const currentTokens = useAtomValue(tokensAtom);
+    const account = useAtomValue(selectedAccountAtom);
```

Comment:
It seems like all of these atoms are mostly used for mapping stuff from one type to another. I feel like this could be simplified a lot by making atoms that specifically target the needs of this component, which seems to be limited to:

- tokens for selected account 
- adding tokens and their metadata to stored tokens

Could encapsulate this in a single atom, and use that instead.